shell.nix

```diff
@@ -0,0 +1,11 @@
+{ pkgs ? import (fetchTarball "http://nixos.org/channels/nixos-22.11/nixexprs.tar.xz") {} }:
+
+pkgs.mkShell {
+  buildInputs = [
+    pkgs.gnumake
+    pkgs.ruby_3_1
+    pkgs.docker
+    pkgs.nodejs
+    pkgs.nodePackages.pnpm
+  ];
+}
```

Comment:
This works but what is it doing? Is it the most optimal thing you can do? What else can/should go in here?